frontend/src/metabase/actions/components/ActionForm/ActionForm.tsx

```diff
@@ -175,15 +175,19 @@ export const ActionForm = ({
       enableReinitialize
     >
       {({ dirty }) => (
-        <Form disabled={!dirty} role="form" data-testid="action-form">
+        <Form
+          disabled={!dirty && !!form.fields.length}
+          role="form"
+          data-testid="action-form"
+        >
```

Comment:
I need a moment to parse `!!form.fields.length`, should we maybe extract it to something like `const hasFields = form.fields.length > 0`? And should we render the form at all if there are no fields?